From: Andy Gospodarek <gospo@redhat.com>
Date: Wed, 5 Nov 2008 16:32:00 -0500
Subject: [net] bonding: update docs for arp_ip_target behavior
Message-id: 20081105213200.GK3714@gospo.rdu.redhat.com
O-Subject: [RHEL5.3 PATCH] bonding: update docs to correctly reflect arp_ip_target behavior
Bugzilla: 468870
RH-Acked-by: David Miller <davem@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: John W. Linville <linville@redhat.com>

This is a patch I did upstream that needs to roll into RHEL:

    commit 3f8b4b13785c2737413d3241c21c7c86a41535ef
    Author: Andy Gospodarek <andy@greyhouse.net>
    Date:   Wed Oct 22 11:19:48 2008 +0000

        bonding: update docs to correctly reflect arp_ip_target behavior

        This documentation patch hopes to clarify that the '+' was only needed
        for Fedora 7 and Red Hat Enterprise Linux 5.0 and 5.1.  After that the
        IP addreses could be added as a comma separated list just like the
        module option.

This will resolve RHBZ 468870.

diff --git a/Documentation/networking/bonding.txt b/Documentation/networking/bonding.txt
index a0cda06..2c9a7b2 100644
--- a/Documentation/networking/bonding.txt
+++ b/Documentation/networking/bonding.txt
@@ -872,17 +872,19 @@ USERCTL=no
 NETMASK, NETWORK and BROADCAST) to match your network configuration.
 
 	For later versions of initscripts, such as that found with Fedora
-7 and Red Hat Enterprise Linux version 5 (or later), it is possible, and,
-indeed, preferable, to specify the bonding options in the ifcfg-bond0
+7 (or later) and Red Hat Enterprise Linux version 5 (or later), it is possible,
+and, indeed, preferable, to specify the bonding options in the ifcfg-bond0
 file, e.g. a line of the format:
 
-BONDING_OPTS="mode=active-backup arp_interval=60 arp_ip_target=+192.168.1.254"
+BONDING_OPTS="mode=active-backup arp_interval=60 arp_ip_target=192.168.1.254"
 
 	will configure the bond with the specified options.  The options
 specified in BONDING_OPTS are identical to the bonding module parameters
-except for the arp_ip_target field.  Each target should be included as a
-separate option and should be preceded by a '+' to indicate it should be
-added to the list of queried targets, e.g.,
+except for the arp_ip_target field when using versions of initscripts older
+than and 8.57 (Fedora 8) and 8.45.19 (Red Hat Enterprise Linux 5.2).  When
+using older versions each target should be included as a separate option and
+should be preceded by a '+' to indicate it should be added to the list of
+queried targets, e.g.,
 
 	arp_ip_target=+192.168.1.1 arp_ip_target=+192.168.1.2
 
@@ -890,7 +892,7 @@ added to the list of queried targets, e.g.,
 options via BONDING_OPTS, it is not necessary to edit /etc/modules.conf or
 /etc/modprobe.conf.
 
-	For older versions of initscripts that do not support
+	For even older versions of initscripts that do not support
 BONDING_OPTS, it is necessary to edit /etc/modules.conf (or
 /etc/modprobe.conf, depending upon your distro) to load the bonding module
 with your desired options when the bond0 interface is brought up.  The
